operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/KubernetesDependentResourceController.java

```diff
@@ -46,45 +43,16 @@ public KubernetesDependentResourceController(DependentResource<R, P> delegate,
             configuration.getDependentResourceClass());
   }
 
-  @SuppressWarnings("unchecked")
-  @Override
-  protected Persister<R, P> initPersister(DependentResource<R, P> delegate) {
-    return (delegate instanceof Persister) ? (Persister<R, P>) delegate : this;
-  }
-
-  @Override
-  public String descriptionFor(R resource) {
-    return String.format("'%s' %s dependent in namespace %s", resource.getMetadata().getName(),
-        resource.getFullResourceName(),
-        resource.getMetadata().getNamespace());
-  }
-
-  @Override
-  public EventSource initEventSource(EventSourceContext<P> context) {
-    this.client = context.getClient();
-    informer = new InformerEventSource<>(configuration, context);
-    return informer;
-  }
-
   @Override
-  public void createOrReplace(R dependentResource, Context context) {
-    client.resource(dependentResource).createOrReplace();
+  public Optional<EventSource> eventSource(EventSourceContext<P> context) {
+    var informer = new InformerEventSource<>(configuration, context);
+    ((KubernetesDependentResource) delegate).withInformerEventSource(informer);
```

Comment:
Maybe we should parameterize the super class with the dependent resource class so that the delegate would always be of the proper type and avoid having to cast? It would also prevent people from creating a `KubernetesDependentResourceController` and passing it a non Kubernetes `DependentResource`…

Reply:
Added a coment will revisit this.